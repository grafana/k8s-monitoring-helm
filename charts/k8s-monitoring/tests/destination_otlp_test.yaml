# yamllint disable rule:document-start rule:line-length rule:trailing-spaces
suite: Destinations - OTLP
templates:
  - alloy-config.yaml
tests:
  - it: creates the Alloy components for an OTLP destination
    set:
      cluster: {name: test-cluster}
      alloy-singleton: {enabled: true, extraConfig: " ", includeDestinations: ["test"]}
      selfReporting: {enabled: false}
      destinations:
        - name: test
          type: otlp
          url: https://otlp.example.com
    asserts:
      - isKind:
          of: ConfigMap
      - matchSnapshot:
          path: data["config.alloy"]

  - it: allows you to disable retry on failure
    set:
      cluster: {name: test-cluster}
      alloy-singleton: {enabled: true, extraConfig: " ", includeDestinations: ["test"]}
      selfReporting: {enabled: false}
      destinations:
        - name: test
          type: otlp
          url: https://otlp.example.com
          retryOnFailure:
            enabled: false
    asserts:
      - isKind:
          of: ConfigMap
      - matchSnapshot:
          path: data["config.alloy"]
