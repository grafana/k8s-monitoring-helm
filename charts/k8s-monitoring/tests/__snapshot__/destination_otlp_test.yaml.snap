allows you to disable retry on failure:
  1: |
    |
      // Destination: test (otlp)
      otelcol.receiver.prometheus "test" {
        output {
          metrics = [otelcol.processor.attributes.test.input]
        }
      }
      otelcol.receiver.loki "test" {
        output {
          logs = [otelcol.processor.attributes.test.input]
        }
      }

      otelcol.processor.attributes "test" {
        output {
          metrics = [otelcol.processor.transform.test.input]
          logs = [otelcol.processor.transform.test.input]
          traces = [otelcol.processor.transform.test.input]
        }
      }

      otelcol.processor.transform "test" {
        error_mode = "ignore"
        metric_statements {
          context = "resource"
          statements = [
            `set(attributes["cluster"], "test-cluster")`,
            `set(attributes["k8s.cluster.name"], "test-cluster")`,
          ]
        }

        metric_statements {
          context = "datapoint"
          statements = [
            `set(attributes["cluster"], "test-cluster")`,
            `set(attributes["k8s.cluster.name"], "test-cluster")`,
            `set(resource.attributes["deployment.environment"], attributes["deployment_environment"] ) where resource.attributes["deployment.environment"] == nil and attributes["deployment_environment"] != nil`,
            `delete_key(attributes, "deployment_environment") where attributes["deployment_environment"] == resource.attributes["deployment.environment"]`,
            `set(resource.attributes["deployment.environment.name"], attributes["deployment_environment_name"] ) where resource.attributes["deployment.environment.name"] == nil and attributes["deployment_environment_name"] != nil`,
            `delete_key(attributes, "deployment_environment_name") where attributes["deployment_environment_name"] == resource.attributes["deployment.environment.name"]`,
            `set(resource.attributes["service.name"], attributes["service_name"] ) where resource.attributes["service.name"] == nil and attributes["service_name"] != nil`,
            `delete_key(attributes, "service_name") where attributes["service_name"] == resource.attributes["service.name"]`,
            `set(resource.attributes["service.namespace"], attributes["service_namespace"] ) where resource.attributes["service.namespace"] == nil and attributes["service_namespace"] != nil`,
            `delete_key(attributes, "service_namespace") where attributes["service_namespace"] == resource.attributes["service.namespace"]`,
          ]
        }
        log_statements {
          context = "resource"
          statements = [
            `set(attributes["cluster"], "test-cluster")`,
            `set(attributes["k8s.cluster.name"], "test-cluster")`,
          ]
        }

        log_statements {
          context = "log"
          statements = [
            `delete_key(attributes, "loki.attribute.labels")`,
            `delete_key(attributes, "loki.resource.labels")`,
            `set(resource.attributes["k8s.container.name"], attributes["container"] ) where resource.attributes["k8s.container.name"] == nil and attributes["container"] != nil`,
            `delete_key(attributes, "container") where attributes["container"] == resource.attributes["k8s.container.name"]`,
            `set(resource.attributes["k8s.cronjob.name"], attributes["cronjob"] ) where resource.attributes["k8s.cronjob.name"] == nil and attributes["cronjob"] != nil`,
            `delete_key(attributes, "cronjob") where attributes["cronjob"] == resource.attributes["k8s.cronjob.name"]`,
            `set(resource.attributes["k8s.daemonset.name"], attributes["daemonset"] ) where resource.attributes["k8s.daemonset.name"] == nil and attributes["daemonset"] != nil`,
            `delete_key(attributes, "daemonset") where attributes["daemonset"] == resource.attributes["k8s.daemonset.name"]`,
            `set(resource.attributes["k8s.deployment.name"], attributes["deployment"] ) where resource.attributes["k8s.deployment.name"] == nil and attributes["deployment"] != nil`,
            `delete_key(attributes, "deployment") where attributes["deployment"] == resource.attributes["k8s.deployment.name"]`,
            `set(resource.attributes["deployment.environment"], attributes["deployment_environment"] ) where resource.attributes["deployment.environment"] == nil and attributes["deployment_environment"] != nil`,
            `delete_key(attributes, "deployment_environment") where attributes["deployment_environment"] == resource.attributes["deployment.environment"]`,
            `set(resource.attributes["deployment.environment.name"], attributes["deployment_environment_name"] ) where resource.attributes["deployment.environment.name"] == nil and attributes["deployment_environment_name"] != nil`,
            `delete_key(attributes, "deployment_environment_name") where attributes["deployment_environment_name"] == resource.attributes["deployment.environment.name"]`,
            `set(resource.attributes["k8s.job.name"], attributes["job_name"] ) where resource.attributes["k8s.job.name"] == nil and attributes["job_name"] != nil`,
            `delete_key(attributes, "job_name") where attributes["job_name"] == resource.attributes["k8s.job.name"]`,
            `set(resource.attributes["k8s.namespace.name"], attributes["namespace"] ) where resource.attributes["k8s.namespace.name"] == nil and attributes["namespace"] != nil`,
            `delete_key(attributes, "namespace") where attributes["namespace"] == resource.attributes["k8s.namespace.name"]`,
            `set(resource.attributes["k8s.pod.name"], attributes["pod"] ) where resource.attributes["k8s.pod.name"] == nil and attributes["pod"] != nil`,
            `delete_key(attributes, "pod") where attributes["pod"] == resource.attributes["k8s.pod.name"]`,
            `set(resource.attributes["k8s.replicaset.name"], attributes["replicaset"] ) where resource.attributes["k8s.replicaset.name"] == nil and attributes["replicaset"] != nil`,
            `delete_key(attributes, "replicaset") where attributes["replicaset"] == resource.attributes["k8s.replicaset.name"]`,
            `set(resource.attributes["service.name"], attributes["service_name"] ) where resource.attributes["service.name"] == nil and attributes["service_name"] != nil`,
            `delete_key(attributes, "service_name") where attributes["service_name"] == resource.attributes["service.name"]`,
            `set(resource.attributes["service.namespace"], attributes["service_namespace"] ) where resource.attributes["service.namespace"] == nil and attributes["service_namespace"] != nil`,
            `delete_key(attributes, "service_namespace") where attributes["service_namespace"] == resource.attributes["service.namespace"]`,
            `set(resource.attributes["k8s.statefulset.name"], attributes["statefulset"] ) where resource.attributes["k8s.statefulset.name"] == nil and attributes["statefulset"] != nil`,
            `delete_key(attributes, "statefulset") where attributes["statefulset"] == resource.attributes["k8s.statefulset.name"]`,
          ]
        }

        trace_statements {
          context = "resource"
          statements = [
            `set(attributes["cluster"], "test-cluster")`,
            `set(attributes["k8s.cluster.name"], "test-cluster")`,
          ]
        }

        output {
          metrics = [otelcol.processor.batch.test.input]
          logs = [otelcol.processor.batch.test.input]
          traces = [otelcol.processor.batch.test.input]
        }
      }

      otelcol.processor.batch "test" {
        timeout = "2s"
        send_batch_size = 8192
        send_batch_max_size = 0

        output {
          metrics = [otelcol.exporter.otlp.test.input]
          logs = [otelcol.exporter.otlp.test.input]
          traces = [otelcol.exporter.otlp.test.input]
        }
      }
      otelcol.exporter.otlp "test" {
        client {
          endpoint = "https://otlp.example.com"
          tls {
            insecure = false
            insecure_skip_verify = false
          }
        }

        retry_on_failure {
          enabled = false
          initial_interval = "5s"
          max_interval = "30s"
          max_elapsed_time = "5m"
        }
      }
creates the Alloy components for an OTLP destination:
  1: |
    |
      // Destination: test (otlp)
      otelcol.receiver.prometheus "test" {
        output {
          metrics = [otelcol.processor.attributes.test.input]
        }
      }
      otelcol.receiver.loki "test" {
        output {
          logs = [otelcol.processor.attributes.test.input]
        }
      }

      otelcol.processor.attributes "test" {
        output {
          metrics = [otelcol.processor.transform.test.input]
          logs = [otelcol.processor.transform.test.input]
          traces = [otelcol.processor.transform.test.input]
        }
      }

      otelcol.processor.transform "test" {
        error_mode = "ignore"
        metric_statements {
          context = "resource"
          statements = [
            `set(attributes["cluster"], "test-cluster")`,
            `set(attributes["k8s.cluster.name"], "test-cluster")`,
          ]
        }

        metric_statements {
          context = "datapoint"
          statements = [
            `set(attributes["cluster"], "test-cluster")`,
            `set(attributes["k8s.cluster.name"], "test-cluster")`,
            `set(resource.attributes["deployment.environment"], attributes["deployment_environment"] ) where resource.attributes["deployment.environment"] == nil and attributes["deployment_environment"] != nil`,
            `delete_key(attributes, "deployment_environment") where attributes["deployment_environment"] == resource.attributes["deployment.environment"]`,
            `set(resource.attributes["deployment.environment.name"], attributes["deployment_environment_name"] ) where resource.attributes["deployment.environment.name"] == nil and attributes["deployment_environment_name"] != nil`,
            `delete_key(attributes, "deployment_environment_name") where attributes["deployment_environment_name"] == resource.attributes["deployment.environment.name"]`,
            `set(resource.attributes["service.name"], attributes["service_name"] ) where resource.attributes["service.name"] == nil and attributes["service_name"] != nil`,
            `delete_key(attributes, "service_name") where attributes["service_name"] == resource.attributes["service.name"]`,
            `set(resource.attributes["service.namespace"], attributes["service_namespace"] ) where resource.attributes["service.namespace"] == nil and attributes["service_namespace"] != nil`,
            `delete_key(attributes, "service_namespace") where attributes["service_namespace"] == resource.attributes["service.namespace"]`,
          ]
        }
        log_statements {
          context = "resource"
          statements = [
            `set(attributes["cluster"], "test-cluster")`,
            `set(attributes["k8s.cluster.name"], "test-cluster")`,
          ]
        }

        log_statements {
          context = "log"
          statements = [
            `delete_key(attributes, "loki.attribute.labels")`,
            `delete_key(attributes, "loki.resource.labels")`,
            `set(resource.attributes["k8s.container.name"], attributes["container"] ) where resource.attributes["k8s.container.name"] == nil and attributes["container"] != nil`,
            `delete_key(attributes, "container") where attributes["container"] == resource.attributes["k8s.container.name"]`,
            `set(resource.attributes["k8s.cronjob.name"], attributes["cronjob"] ) where resource.attributes["k8s.cronjob.name"] == nil and attributes["cronjob"] != nil`,
            `delete_key(attributes, "cronjob") where attributes["cronjob"] == resource.attributes["k8s.cronjob.name"]`,
            `set(resource.attributes["k8s.daemonset.name"], attributes["daemonset"] ) where resource.attributes["k8s.daemonset.name"] == nil and attributes["daemonset"] != nil`,
            `delete_key(attributes, "daemonset") where attributes["daemonset"] == resource.attributes["k8s.daemonset.name"]`,
            `set(resource.attributes["k8s.deployment.name"], attributes["deployment"] ) where resource.attributes["k8s.deployment.name"] == nil and attributes["deployment"] != nil`,
            `delete_key(attributes, "deployment") where attributes["deployment"] == resource.attributes["k8s.deployment.name"]`,
            `set(resource.attributes["deployment.environment"], attributes["deployment_environment"] ) where resource.attributes["deployment.environment"] == nil and attributes["deployment_environment"] != nil`,
            `delete_key(attributes, "deployment_environment") where attributes["deployment_environment"] == resource.attributes["deployment.environment"]`,
            `set(resource.attributes["deployment.environment.name"], attributes["deployment_environment_name"] ) where resource.attributes["deployment.environment.name"] == nil and attributes["deployment_environment_name"] != nil`,
            `delete_key(attributes, "deployment_environment_name") where attributes["deployment_environment_name"] == resource.attributes["deployment.environment.name"]`,
            `set(resource.attributes["k8s.job.name"], attributes["job_name"] ) where resource.attributes["k8s.job.name"] == nil and attributes["job_name"] != nil`,
            `delete_key(attributes, "job_name") where attributes["job_name"] == resource.attributes["k8s.job.name"]`,
            `set(resource.attributes["k8s.namespace.name"], attributes["namespace"] ) where resource.attributes["k8s.namespace.name"] == nil and attributes["namespace"] != nil`,
            `delete_key(attributes, "namespace") where attributes["namespace"] == resource.attributes["k8s.namespace.name"]`,
            `set(resource.attributes["k8s.pod.name"], attributes["pod"] ) where resource.attributes["k8s.pod.name"] == nil and attributes["pod"] != nil`,
            `delete_key(attributes, "pod") where attributes["pod"] == resource.attributes["k8s.pod.name"]`,
            `set(resource.attributes["k8s.replicaset.name"], attributes["replicaset"] ) where resource.attributes["k8s.replicaset.name"] == nil and attributes["replicaset"] != nil`,
            `delete_key(attributes, "replicaset") where attributes["replicaset"] == resource.attributes["k8s.replicaset.name"]`,
            `set(resource.attributes["service.name"], attributes["service_name"] ) where resource.attributes["service.name"] == nil and attributes["service_name"] != nil`,
            `delete_key(attributes, "service_name") where attributes["service_name"] == resource.attributes["service.name"]`,
            `set(resource.attributes["service.namespace"], attributes["service_namespace"] ) where resource.attributes["service.namespace"] == nil and attributes["service_namespace"] != nil`,
            `delete_key(attributes, "service_namespace") where attributes["service_namespace"] == resource.attributes["service.namespace"]`,
            `set(resource.attributes["k8s.statefulset.name"], attributes["statefulset"] ) where resource.attributes["k8s.statefulset.name"] == nil and attributes["statefulset"] != nil`,
            `delete_key(attributes, "statefulset") where attributes["statefulset"] == resource.attributes["k8s.statefulset.name"]`,
          ]
        }

        trace_statements {
          context = "resource"
          statements = [
            `set(attributes["cluster"], "test-cluster")`,
            `set(attributes["k8s.cluster.name"], "test-cluster")`,
          ]
        }

        output {
          metrics = [otelcol.processor.batch.test.input]
          logs = [otelcol.processor.batch.test.input]
          traces = [otelcol.processor.batch.test.input]
        }
      }

      otelcol.processor.batch "test" {
        timeout = "2s"
        send_batch_size = 8192
        send_batch_max_size = 0

        output {
          metrics = [otelcol.exporter.otlp.test.input]
          logs = [otelcol.exporter.otlp.test.input]
          traces = [otelcol.exporter.otlp.test.input]
        }
      }
      otelcol.exporter.otlp "test" {
        client {
          endpoint = "https://otlp.example.com"
          tls {
            insecure = false
            insecure_skip_verify = false
          }
        }

        retry_on_failure {
          enabled = true
          initial_interval = "5s"
          max_interval = "30s"
          max_elapsed_time = "5m"
        }
      }
